synapse/federation/sender/__init__.py

```diff
@@ -512,7 +511,9 @@ async def handle_room_events(events: List[EventBase]) -> None:
                             await handle_event(event)
 
                 events_by_room: Dict[str, List[EventBase]] = {}
-                for event in events:
+
+                for event_cache in event_entries.values():
+                    event = event_cache.event
                     events_by_room.setdefault(event.room_id, []).append(event)
```

Comment:
Isn't this still making assumptions about the order of the result of `get_unredacted_events_from_cache_or_db` ?

Reply:
:facepalm: Right, I totally forgot that was why I needed to iterate over `event_to_received_ts`. I've fixed this, sorry for being a dunce.

Reply:
:laughing: no worries, easily done!